torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3337,10 +3337,14 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
       fn(tensor, comm_, ncclStream, p2pTargetRank),
       ncclComm->getNcclCommFailureReason());
 #else
-  C10D_NCCL_CHECK_TIMEOUT(
-      fn(tensor, comm_, ncclStream, p2pTargetRank),
-      ncclComm->getNcclComm(),
-      ncclComm->getNcclCommFailureReason());
+  // In non-blocking mode, we need to use ncclGroup semantics to ensure that the
+  // kernel is enqueued for single-P2P ops.  Otherwise, the event record below
+  // may not capture the kernel, leading to data corruption.
+  ncclGroupStart();
+  C10D_NCCL_CHECK_NONBLOCKING(
```

Comment:
Should this MACRO be applied when non-blocking mode is on? otherwise we should just use           C10D_NCCL_CHECK

Reply:
I don't quite understand the question. We are in the `NCCL_HAS_COMM_NONBLOCKING` region, and we are using the `C10D_NCCL_CHECK_NONBLOCKING` macro.

Reply:
Sorry, I mean why not use C10D_NCCL_CHECK_TIMEOUT to wait for ncclSuccess, same as those used in other  collectives

Reply:
Oh, the TIMEOUT check is deferred to the `ncclGroupEnd()` call. I think that's more efficient.

Reply:
i.e. we don't have timeout concern for calls within group semantics.